src/components/UserSelector.tsx

```diff
@@ -1,17 +1,35 @@
-import React from 'react';
+import { useState } from "react";
+import cn from 'classnames';
+
+import { User } from "../types/User";
+import { OneUser } from "./OneUser";
+
+type Props = {
+  users: User[];
+  setSelectedUserId: (id: number | null) => void;
+  selectedUserId: number | null;
+};
+
+export const UserSelector: React.FC<Props> = ({ users, setSelectedUserId, selectedUserId }) => {
+  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+  
+  const activeUser = users.find(user => user.id === selectedUserId);
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div data-cy="UserSelector" className={cn("dropdown", {"is-active": isDropDownOpen}) } >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsDropDownOpen(!isDropDownOpen)}
         >
-          <span>Choose a user</span>
-
+          {activeUser?.name
```

Comment:
Ensure that the `activeUser` variable correctly handles cases where `selectedUserId` does not match any user in the `users` array. This will prevent potential issues with undefined values.